include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef VIGRA_NO_INT_ISNAN
```

Comment:
I'm leaning to the conclusion that it is much better to change `rf_preprocessing.hxx` to call
```
      if(isnan(NumericTraits<T>::toRealPromote(in[ii])))
```
to solve the problem where it occurs.